### PR TITLE
🐳 Docker container clean up after ingest ends

### PR DESCRIPTION
**Pull request by znatty22**
**Problem**
Docker container runs as root user or a user different than Jenkins user. When we run the container in the kf-ingest-packages CI pipeline on Jenkins host, we need to bind mount the kf-ingest-packages/kf_ingest_packages/packages directory into the container. When container is running ingest it writes logs/outputs as root into that bind mounted directory. When Jenkins tries to do its usual clean up it tries to delete the kf-ingest-packages/packages directory which contains logs/outputs that are owned by root.

**Solution**
Update docker container entrypoint so that it optionally cleans up (rm -rf) the ingest package dir after ingest finishes or fails. This is controlled by the CLEANUP env var which will contain the package name to delete. This is used in the CI script for kf-ingest-packages.

### CODE REVIEW

File: tests/data/SD_ME0WME0W/ingest_package_config.py [hunk 1]
```diff
@@ -19,6 +19,11 @@
     'read_group_genomic_file'
 ]
 
+expected_counts = {
```
Comment (fiendish) [inline tests/data/SD_ME0WME0W/ingest_package_config.py]: unrelated

File: scripts/entrypoint.sh [hunk 2]
```diff
@@ -22,4 +22,26 @@ if [[ $KF_INGEST_APP_MODE = "production" ]]; then
...
+    INGEST_ERROR=1
+fi
+
+# During deployment - rm ingest package after ingest completes
+if [[ $CLEANUP ]]; then
+    INGEST_PACKAGE_DIR="/data/packages/$CLEANUP"
+    if [[ -d $INGEST_PACKAGE_DIR ]]; then
+        echo "Start cleanup ..."
+        echo "Deleting ingest package $INGEST_PACKAGE_DIR"
+        rm -rf $INGEST_PACKAGE_DIR
```
Comment (fiendish) [inline scripts/entrypoint.sh]: please quote this variable and put either `set -u` or `set -o nounset` at the start of the file

Comment (znatty22) [reply scripts/entrypoint.sh]: k, will update

Comment (fiendish) [reply scripts/entrypoint.sh]: Also, what if the script wasn't run by docker?

Comment (fiendish) [reply scripts/entrypoint.sh]: Consider...

```
#!/bin/bash

mkdir -p "test"
CLEANUP="foo bar"
touch "test/$CLEANUP"
echo "CLEANING $CLEANUP"
ND="test/$CLEANUP"
rm $ND
echo "done"
```

prints:
> CLEANING foo bar
> rm: test/foo: No such file or directory
> rm: bar: No such file or directory
> done

vs 

```
#!/bin/bash

mkdir -p "test"
CLEANUP="foo bar"
touch "test/$CLEANUP"
echo "CLEANING $CLEANUP"
ND="test/$CLEANUP"
rm "$ND"
echo "done"
```

prints:
> CLEANING foo bar
> done

Comment (fiendish) [reply scripts/entrypoint.sh]: I think that CLEANUP should be passed in as an argument to the script as possible and the check should not be based on whether the environment has something set.
